### PR TITLE
[feature](cloud) Support decomission in cloud

### PR DESCRIPTION
**Pull request by deardeng**
## Proposed changes

Issue Number: close #xxx

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by deardeng** [comment]
run buildall

**Comment by deardeng** [comment]
run buildall

**Review by yujun777 — APPROVED** [review]
LGTM

**Comment by deardeng** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/catalog/Database.java [hunk 1]
```diff
@@ -470,6 +470,10 @@ public Map<Long, Table> getIdToTableRef() {
         return idToTable;
     }
 
+    public List<Long> getTableIds() {
+        return new ArrayList<>(idToTable.keySet());
```
Comment (gavinchou) [inline fe/fe-core/src/main/java/org/apache/doris/catalog/Database.java:474]: Do we need to lock and make a copy of table ids here?

File: fe/fe-core/src/main/java/org/apache/doris/alter/SystemHandler.java [hunk 2]
```diff
@@ -78,7 +78,9 @@ public SystemHandler() {
     @Override
     protected void runAfterCatalogReady() {
         super.runAfterCatalogReady();
-        runAlterJobV2();
+        if (Config.isNotCloudMode()) {
+            runAlterJobV2();
```
Comment (gavinchou) [inline fe/fe-core/src/main/java/org/apache/doris/alter/SystemHandler.java:82]: Why? Pls dd a comment if there is a specific reason

File: fe/fe-common/src/main/java/org/apache/doris/common/Config.java [hunk 3]
```diff
@@ -2721,7 +2721,7 @@ public static boolean isNotCloudMode() {
     public static int cloud_min_balance_tablet_num_per_run = 2;
 
     @ConfField(mutable = true, masterOnly = true)
-    public static boolean cloud_preheating_enabled = false;
+    public static boolean cloud_preheating_enabled = true;
```
Comment (gavinchou) [inline fe/fe-common/src/main/java/org/apache/doris/common/Config.java:2724]: Unify naming  to xxx_warm_up_xxx.
e.g. enable_warm_up_for_rebalance

**Review by gavinchou — CHANGES_REQUESTED** [review]

File: fe/fe-common/src/main/java/org/apache/doris/common/Config.java (hunk 3 above)
Comment (deardeng) [reply fe/fe-common/src/main/java/org/apache/doris/common/Config.java:2724]: 这开关是预热的啊

Comment (gavinchou) [reply fe/fe-common/src/main/java/org/apache/doris/common/Config.java:2724]: all "pre heat" and "pre cache" have been and should be named as "xxx_warm_up_xxx"
the previous names are confusing and not unified.

File: fe/fe-core/src/main/java/org/apache/doris/catalog/Database.java (hunk 1 above)
Comment (deardeng) [reply fe/fe-core/src/main/java/org/apache/doris/catalog/Database.java:474]: 不用吧，idToTable 是个ConcurrentMap，线程安全的。而且line 473 附近几个idToTable 也都没加锁

File: fe/fe-core/src/main/java/org/apache/doris/alter/SystemHandler.java (hunk 2 above)
Comment (deardeng) [reply fe/fe-core/src/main/java/org/apache/doris/alter/SystemHandler.java:82]: // check all decommissioned backends, if there is no available tablet on that backend, drop it.
    private void runAlterJobV2() 

这个逻辑和存算分离的decomission 逻辑冲突

File: fe/fe-common/src/main/java/org/apache/doris/common/Config.java (hunk 3 above)
Comment (deardeng) [reply fe/fe-common/src/main/java/org/apache/doris/common/Config.java:2724]: fixed

**Comment by dataroaring** [comment]
run buildall

**Review by gavinchou — APPROVED** [review]

**Review by dataroaring — APPROVED** [review]
LGTM

**Pull request merged**